src/main/java/thestonedturtle/partypanel/ui/PlayerPanel.java

```diff
@@ -172,8 +214,18 @@ public void changePlayer(final PartyPlayer newPlayer)
 				}
 
 				final SkillPanelSlot panel = skillsPanel.getPanelMap().get(s);
+				int newExp = player.getStats().getSkillEXPs().get(s);
+				int nextLevelExp = Experience.getXpForLevel(player.getStats().getBaseLevels().get(s)+1);
```

Comment:
You are not handling the potential `IllegalArgumentException` this method will throw if someone is already level 126 in a skill, which will make this pass `127` which is `> MAX_VIRT_LEVEL`.

I'd recommend removing this line and replacing the `tooltipExp` code block with this:
```java
String tooltipExp = "<html>" + s.getName() + "<br/>";
tooltipExp += "XP: " + NumberFormat.getNumberInstance().format(newExp);

int curLevel = player.getStats().getBaseLevels().get(s);
if (curLevel > 0 && curLevel < 126)
{
	int nextLevelExp = Experience.getXpForLevel(curLevel + 1);
	tooltipExp += "<br/>Level in: " + NumberFormat.getNumberInstance().format(nextLevelExp - newExp) + "</html>";
}
```

Reply:
Great point, thanks for catching that!